source/frontend-testing.rst

```diff
@@ -0,0 +1,334 @@
+Front-end Testing Practices
+###########################
+
+This documentation reflects the present state of our knowledge of how to test
+front-end JavaScript code.
+
+It also reflects a series of choices that were basically arbitrary because of
+the lack of anything like a standard or clear best choice (e.g. the
+decision to use one test runner or assertion library over alternatives).
+
+These choices are still being experimented with, and developers should feel
+free to suggest revisions if they're found to perform poorly in practice.
+
+
+Testing libraries: overview
+###########################
+
+Mocha
+-----
+
+`Mocha <https://mochajs.org/#features>`_ is a simple and modular test library.
+Unlike `Jasmine <http://jasmine.github.io/>`_, Mocha doesn't come with an
+assertion library, facilities for mocks and spies, and so on. Instead it provides
+a framework for writing and executing tests and lets you choose other libraries
+to meet those needs.
+
+To create a Mocha test that will be recognized by our Gulp build process,
+create a file with the extension ``.spec.js`` and place it in a ``test/`` subdir
```

Comment:
Would it be unreasonable to change this pattern from `*.spec.js` to `test_*.js` just so it is more consistent with the existing backend tests?